app/src/org/commcare/views/TabbedDetailView.java

```diff
@@ -61,79 +61,43 @@ private void loadViewConfig(Context context, AttributeSet attrs) {
+        mContext = (AppCompatActivity) context;
     }
 
     /*
      * Attach this view to a layout.
      */
     public void setRoot(ViewGroup root) {
-        LayoutInflater inflater = (LayoutInflater)mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+        LayoutInflater inflater = (LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
 
         inflater.inflate(R.layout.tabbed_detail_view, root, true);
 
-        mMenu = root.findViewById(R.id.tabbed_detail_menu);
         mViewPager = root.findViewById(R.id.tabbed_detail_pager);
         mViewPager.setId(AndroidUtil.generateViewId());
-
-        mViewPageTabStrip = root.findViewById(R.id.pager_tab_strip);
-
-        mViewPager.setOnPageChangeListener(new ViewPager.OnPageChangeListener() {
-
-            @Override
-            public void onPageSelected(int position) {
-                markSelectedTab(position);
-            }
-
-            @Override
-            public void onPageScrolled(int arg0, float arg1, int arg2) {
-            }
-
-            @Override
-            public void onPageScrollStateChanged(int arg0) {
-            }
-
-        });
-    }
-
-    public void showMenu() {
-        mMenu.setVisibility(VISIBLE);
+        mTabLayout = root.findViewById(R.id.tab_layout);
     }
 
     /**
      * Get form list from database and insert into view.
      */
     public void refresh(Detail detail, TreeReference reference, int index) {
-        EntityDetailPagerAdapter entityDetailPagerAdapter =
-                new EntityDetailPagerAdapter(mContext.getSupportFragmentManager(), detail, index, reference,
-                        new ListItemViewStriper(this.mOddColor, this.mEvenColor));
+        EntityDetailPagerAdapter entityDetailPagerAdapter = new EntityDetailPagerAdapter(mContext.getSupportFragmentManager(), mContext.getLifecycle(), detail, index, reference, new ListItemViewStriper(this.mOddColor, this.mEvenColor));
         mViewPager.setAdapter(entityDetailPagerAdapter);
-        if (!detail.isCompound()) {
-            if (mViewPageTabStrip != null) {
-                mViewPageTabStrip.setVisibility(GONE);
-            }
-        }
-    }
-
-    /**
-     * Style one tab as "selected".
-     */
-    private void markSelectedTab(int position) {
-        if (mMenu.getChildCount() <= position) {
-            return;
-        }
 
-        for (int i = 0; i < mMenu.getChildCount(); i++) {
-            mMenu.getChildAt(i).setBackgroundDrawable(getResources().getDrawable(R.drawable.title_neutral_tab_vertical));
+        if (detail.isCompound()) {
+            new TabLayoutMediator(mTabLayout, mViewPager,
+                    (tab, position) -> tab.setText(detail.getDetails()[position].getTitle().getText().evaluate()))
+                    .attach();
+        } else {
+            mTabLayout.setVisibility(GONE);
         }
-        mMenu.getChildAt(position).setBackgroundDrawable(getResources().getDrawable(R.drawable.title_case_tab_vertical));

```

Comment:
why was this removed ?

Reply:
I believe the default style is enough, but will make sure it looks fine everywhere

Reply:
@shubham1g5 Here's the old style to which those drawables where used:
![645ebc7e-74a2-11e4-8de9-cf45e08495a6](https://github.com/user-attachments/assets/6eb7eec2-9418-469b-b427-c10793981b8d)

Reply:
and how does it look now with the changes in PR ?

Reply:
@shubham1g5 my understanding is that this style was used by the old tab system, which was based on the [tabbed_detail_menu](https://github.com/dimagi/commcare-android/blob/423846919137930ca0610e268c2ba8a0f60177be/app/res/layout/tabbed_detail_view.xml#L7C26-L7C44) linear layout. That system was discontinued when `PagerTabStrip` was added to CC, [this](https://github.com/dimagi/commcare-android/pull/401/files#diff-2e88d3fe8ebeb3aa7b671fbeeae867da2e9af67242f2d2e227a38a465b835679) is the PR. So this PR doesn't change the style around the tabs, the current and previous styling is what I shared in [this](https://github.com/dimagi/commcare-android/pull/2873#discussion_r1890009087) comment.

Reply:
I think my question here is more around how does this change not effect Case Detail tabs for example given `TabbedDetailView` is used there and this line of code seems generic to all `TabbedDetailView` layouts.

Reply:
hannn, in the old tab system, we used to add `HorizontalMediaView` views to [tabbed_detail_menu](https://github.com/dimagi/commcare-android/blob/423846919137930ca0610e268c2ba8a0f60177be/app/res/layout/tabbed_detail_view.xml#L7C26-L7C44) linear layout and apply this style to them, this was removed [here](https://github.com/dimagi/commcare-android/pull/401/commits/0ce8f12bf0bd7b097d4ca803843c29bfe041b9c9#diff-2e88d3fe8ebeb3aa7b671fbeeae867da2e9af67242f2d2e227a38a465b835679L108). And because we don't add views to that linear layout, every time we call [mMenu.getChildCount()](https://github.com/dimagi/commcare-android/blob/423846919137930ca0610e268c2ba8a0f60177be/app/src/org/commcare/views/TabbedDetailView.java#L122C9-L122C49) we get 0 and the execution is interrupted. This to say that we haven't been calling this code for quite some time, it's dead code now.

Reply:
got it , thanks for confirming.